libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/AztecEditorFragment.java

```diff
@@ -405,9 +406,17 @@ public void appendMediaFile(final MediaFile mediaFile, final String mediaUrl, Im
                 attrs.addAttribute("", "data-wpid", "data-wpid", "string", localMediaId);
                 attrs.addAttribute("", "src", "src", "string", safeMediaUrl);
 
-                Bitmap bitmap = BitmapFactory.decodeFile(safeMediaUrl);
-
-                content.insertMedia(new BitmapDrawable(getResources(), bitmap), attrs);
+                // load a scaled version of the image to prevent OOM exception
+                int maxWidth = DisplayUtils.getDisplayPixelWidth(getActivity());
+                Bitmap bitmapToShow = ImageUtils.getWPImageSpanThumbnailFromFilePath(getActivity(), safeMediaUrl, maxWidth);
+                if (bitmapToShow == null) {
+                    ToastUtils.showToast(getActivity(), R.string.error_media_load);
+                    try {
+                        bitmapToShow = BitmapFactory.decodeResource(getResources(), R.drawable.media_image_placeholder);
```

Comment:
The exception handler here doesn't appear necessary since `Bitmap.decodeResource()` catches all exceptions. There is a chance, however, that `Bitmap.decodeResource()` will return `null` so maybe we want to handle that?